routing/control_tower.go

```diff
@@ -107,46 +111,61 @@ func (p *controlTower) RegisterAttempt(paymentHash lntypes.Hash,
 func (p *controlTower) Success(paymentHash lntypes.Hash,
 	preimage lntypes.Preimage) error {
 
-	route, err := p.db.Success(paymentHash, preimage)
+	payment, err := p.db.Success(paymentHash, preimage)
 	if err != nil {
 		return err
 	}
 
 	// Notify subscribers of success event.
 	p.notifyFinalEvent(
-		paymentHash, createSuccessResult(route, preimage),
+		paymentHash, createSuccessResult(payment.HTLCs),
 	)
 
 	return nil
 }
 
 // createSuccessResult creates a success result to send to subscribers.
-func createSuccessResult(rt *route.Route,
-	preimage lntypes.Preimage) *PaymentResult {
+func createSuccessResult(htlcs []channeldb.HTLCAttempt) *PaymentResult {
+	// Extract any preimage from the list of HTLCs.
+	var (
+		preimage lntypes.Preimage
+		route    *route.Route
```

Comment:
could rather have this single route being extracted by the consumer of the PaymentResult

Reply:
fixed in last commit!